server/src/main/scala/com/cloudera/livy/sessions/SessionManager.scala

```diff
@@ -41,7 +47,12 @@ class SessionManager[S <: Session](val livyConf: LivyConf) extends Logging {
   private[this] final val sessionTimeout =
     TimeUnit.MILLISECONDS.toNanos(livyConf.getTimeAsMs(SessionManager.SESSION_TIMEOUT))
 
+  private val yarnClient = YarnClient.createYarnClient()
```

Comment:
It takes a while to initialize yarn client. Especially when no YARN is running at the default port on the same machine.
I would not initialize yarn client unless Livy is running on YARN.
I implemented these handling in https://github.com/cloudera/livy/pull/163, so if you can wait a bit, you can get these for free.